File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace MiYABiS.Study.インタフェースの扱い.CS
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            _compute(new ComputeAddition());
        }

        private void Button2_Click(object sender, EventArgs e)
        {
            _compute(new ComputeSubtraction());
        }

        private void _compute(ICompute obj)
        {
            try
            {
                txtAns.Text = obj.Execute(long.Parse(txtValue1.Text), long.Parse(txtValue2.Text)).ToString();
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
        }
    }

}

File: VB/Form1.vb

Public Class Form1

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        _compute(New ComputeAddition())
    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        _compute(New ComputeSubtraction())
    End Sub

    Private Sub _compute(ByVal obj As ICompute)
        Try
            txtAns.Text = obj.Execute(CLng(txtValue1.Text), CLng(txtValue2.Text))
        Catch ex As Exception
            MessageBox.Show(ex.Message)
        End Try
    End Sub

End Class
